src/Hazelcast.Net.Tests/Networking/NetworkHostnameTests.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Hazelcast.Testing;
+using NUnit.Framework;
+using Hazelcast.Core;
+using Hazelcast.Testing.Remote;
+using System.Threading;
+
+namespace Hazelcast.Tests.Networking
+{
+    internal class NetworkHostnameTests : RemoteTestBase
+    {
+        public IRemoteControllerClient RcClient { get; set; }
+
+        public Cluster RcCluster { get; set; }
+
+        [SetUp]
+        public async Task ClusterOneTimeSetUp()
+        {
+            // create remote client and cluster
+            RcClient = await ConnectToRemoteControllerAsync().CfAwait();
+            RcCluster = await RcClient.CreateClusterAsync(Hazelcast.Testing.Remote.Resources.hazelcast_hostname).CfAwait();
+        }
+
+        [TearDown]
+        public async Task ClusterOneTimeTearDown()
+        {
+            // terminate & remove client and cluster
+            if (RcClient != null)
+            {
+                if (RcCluster != null)
+                    await RcClient.ShutdownClusterAsync(RcCluster).CfAwait();
+                await RcClient.ExitAsync().CfAwait();
+            }
+        }
+
+        [Test]
+        public async Task ConnectWithDNSHostnames()
```

Comment:
I don't understand what we're trying to test. Maybe a comment is required. Are we testing that `options.Networking.Addresses.Add("localhost");` do work? And then why do we need to test against 2 members? Why do we need a special member configuration file? Are we testing that a *member* can be configured with `localhost` and why is this a test that is performed by clients and not by the core/member code itself?

Reply:
Yes, you are right. It just tests to connect via hostname. I ported them as it is in java. However, I took them from linked old PR but the tests are changed/removed completely on the current version of the master(hazelcast). It's silly me I didn't think that tests could be removed/changed today. I/we should be more carefully about the old issues. I think it is not a valid issue anymore. (Previous comment lost while posting)